Add vexctl package

jspeed-meyers opened the pull request:
This PR packages `vexctl`, a Go tool for creating, transforming, and attesting VEX metadata. It is part of the OpenVEX project.

### Pre-review Checklist

#### For new package PRs only
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`) (no support policy present)

Commit:
Add vexctl package

Signed-off-by: John Speed Meyers <jsmeyers@chainguard.dev>

Commit:
Merge branch 'main' into add-vexctl-package

rawlingsj commented on vexctl.yaml:
```diff
@@ -0,0 +1,32 @@
+  description: A tool to create, transform and attest VEX metadata
+  copyright:
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - go
```
```suggestion
```
We can remove the runtime dependency on go

jspeed-meyers replied:
Will remove. Cooking up a commit! TY

rawlingsj commented on vexctl.yaml:
```diff
@@ -0,0 +1,32 @@
+      - build-base
+      - busybox
+      - ca-certificates-bundle
+      - go
```
```suggestion
```
This is automagically added by the [go/install](https://github.com/chainguard-dev/melange/blob/a4590d130eb51801d2819183e5cdc19fbcd079db/pkg/build/pipelines/go/install.yaml#L5) pipeline so can be removed

jspeed-meyers replied:
Ohhh, too cool. Will remove.

jspeed-meyers replied:
See commit b17865a

rawlingsj reviewed (changes requested):
looks great! just two minor comments

imjasonh commented on vexctl.yaml:
```diff
@@ -0,0 +1,32 @@
+    - license: Apache-2.0
+  dependencies:
+    runtime:
+      - go
```
Just checking: does this actually need go at runtime? Most Go apps don't need Go as a runtime dep, unless maybe they shell out to `go build` or something (like `ko` does).

This is as much an admission of my inexperience about vexctl as anything else.

Commit:
Remove runtime and build dependencies

The runtime dependency is unneeded. The buildtime dependencies are implied by using `go-install`.

Signed-off-by: John Speed Meyers <jsmeyers@chainguard.dev>

jspeed-meyers commented:
Thank you, @rawlingsj and @imjasonh.
